contributors/devel/sig-node/separate-mount-namespace.md

```diff
@@ -0,0 +1,212 @@
+# Managing a Separate Mount Namespace for Kubernetes
+
+## What does this mean?
+
+Originally, Kubelet and the Container Runtime all ran in the same mount namespace as
+the host OS (systemd, login shells, system daemons).
+
+![Fig 1: Original mount propagation](images/original-k8s-mount-propagation.png)
+
+_Fig 1: Original mount propagation_
+
+However, in order to support the full range of cloud-native applications, only
+two guarantees need to be made about the relationship of mountpoints:
+
+1. Mounts originating in the host OS are visible to the Container Runtime, OCI
+   hooks, Kubelet, and container volumeMounts with `MountPropagation:
+   HostToContainer` (or `Bidirectional`)
+2. Mounts originating in the Container Runtime, OCI hooks, Kubelet, and
+   container volumeMounts with `mountPropagation: Bidirectional` are visible to
+   each other and container volumeMounts with `MountPropagation:
+   HostToContainer`
+
+The fact that mounts originating in the Container Runtime, OCI hooks, Kubelet,
+and container volumeMounts with `mountPropagation: Bidirectional` were visible
+to the host OS in the original implementations of Kubernetes was accidental to
+the implementation, and there are a number of real-world benefits to changing
+the way Kubernetes is installed in order to hide the container-specific mount
+points from the host OS.
+
+![Fig 2: Container-Runtime-scoped mount propagation](images/hidden-k8s-mount-propagation.png)
+
+_Fig 2: Container-Runtime-scoped mount propagation_
+
+## Benefits
+
+### Systemd Efficiency
+
+Today systemd's CPU usage is adversely affected by a large number of
+mountpoints, especially versions released prior to December 2020. This feature
+of systemd, which scans all /proc/1/mount to invoke udev events for every
+change suffers from two issues. The first was an inefficiency in systemd’s
+processing logic. This was fixed in December 2020, and this fix reportedly
+reduced the systemd mount-scanning-related overhead by 30-50%. The second is a
+lack of granularity in the events the kernel sends to systemd about mountpoint
+changes, which necessitates a full rescan of all mountpoints even on a minor
+change or single mountpoint addition. The kernel change to remedy this seems
+unlikely to move forward in the near term.  (See
+[BZ1819869](https://bugzilla.redhat.com/show_bug.cgi?id=1819868) for more
+details)
+
+For example, a fairly small OpenShift deployment today (which does not yet have
+the systemd December 2020 fix) has upwards of 400 mountpoints on every node
+owned by Kubelet, the Container Runtime, and infrastructure containers like OVN
+or CSI operators. Hiding the container mountpoints reduces the steady-state
```

Comment:
Do we have an upstream Kubernetes parallel use case we could reference here instead?

Reply:
A great question.  I don't have one, but it may be worth setting on up.